src/bentoml/_internal/io_descriptors/json.py

```diff
@@ -192,6 +200,24 @@ def __init__(
                 pydantic_model, pydantic.BaseModel
             ), "'pydantic_model' must be a subclass of 'pydantic.BaseModel'."
 
+        if not typed_dict and sample:
+            raise BadInput(
+                "JSON argument 'sample' only support with typed_dict if you want to set sample "
+                "use JSON.from_sample()"
+            )
+        if (
+            typed_dict and typed_dict is not dict
+        ):  # there is no way to check isinstance TypedDict in runtime
+            assert issubclass(
+                typed_dict, dict
+            ), "'typed_dict' must be a subclass of 'typing.TypedDict or dict'."
+
+        if typed_dict:
+            from pydantic import create_model_from_typeddict
+
+            self._typed_dict_model = create_model_from_typeddict(typed_dict)
```

Comment:
Does this mean typed_dict depends on pydantic? it seems kinda strange that in order to use TypedDict you need to have pydantic installed.

Reply:
If this is the case then `typed_dict` and `pydantic_model` should be mutually exclusive then.

Might also need a import check or just lazy load it